library/ssl_tls.c

```diff
@@ -6887,6 +7489,20 @@ static int ssl_check_ctr_renegotiate( mbedtls_ssl_context *ssl )
     MBEDTLS_SSL_DEBUG_MSG( 1, ( "record counter limit reached: renegotiate" ) );
     return( mbedtls_ssl_renegotiate( ssl ) );
 }
+
+int mbedtls_ssl_check_renego_not_honored( mbedtls_ssl_context *ssl )
+{
+    ssl->renego_records_seen++;
```

Comment:
The name of this function does not reflect the fact that it also changes the internal ssl state - one may think that it only performs a check.

Reply:
Agreed and changed.